backend/app/treasury/tests/test_covalent.py

```diff
@@ -0,0 +1,327 @@
+# pylint: disable=protected-access
+from copy import deepcopy
+from datetime import datetime, timedelta
+from json import loads
+from json.decoder import JSONDecodeError
+
+from dateutil.parser import parse
+from dateutil.tz import UTC
+from dateutil.utils import today as today_in
+from fakeredis import FakeRedis
+from httpx import AsyncClient
+from pytest import MonkeyPatch, mark, raises
+
+from backend.app.treasury.adapters import covalent
+
+from .conftest import (
+    HTTPStatusError,
+    covalent_portfolio_v2_transfers,
+    covalent_transfers_v2_transfers,
+    raise_http_status_error_404,
+    spam_token_transfer,
+)
+
+
+class MockResponse:
+    status_code = None
+    text = ""
+
+    def raise_for_status(self):
+        return None
+
+    @staticmethod
+    def json():
+        return {
+            "prices": [
+                [1590019200000, 0.9993188217729783],
+                [1590105600000, 0.9981200719764681],
+                [1590192000000, 0.9995106820722109],
+                [1590278400000, 0.9984064931995519],
+                [1590364800000, 0.9941101041414783],
+            ]
+        }
+
+
+async def return_mocked_resp(*_, **__):
+    return MockResponse()
+
+
+def mocked_datetime(offset: int = 0, use_today: bool = False):
+    _mocked_datetime = datetime(2022, 1, 1) if not use_today else today_in(UTC)
+    return _mocked_datetime + timedelta(offset)
+
+
+@mark.asyncio
+async def test_covalent_transfers_success(monkeypatch: MonkeyPatch):
+    monkeypatch.setattr(AsyncClient, "get", return_mocked_resp)
+    monkeypatch.setattr(
+        MockResponse,
+        "json",
+        lambda *_: {
+            "data": {
+                "address": "0x6d6f636b65645f7472656173757279",
+                "quote_currency": "USD",
+                "chain_id": 1,
+                "items": covalent_transfers_v2_transfers,
+            }
+        },
+    )
+
+    treasury_transfers = await covalent.get_transfer_data(
+        "0x6d6f636b65645f7472656173757279", "", 1
+    )
+
+    assert treasury_transfers == {
+        "address": "0x6d6f636b65645f7472656173757279",
+        "quote_currency": "USD",
+        "chain_id": 1,
+        "items": covalent_transfers_v2_transfers,
+    }
+
+
+@mark.asyncio
+async def test_covalent_transfers_status_err(monkeypatch: MonkeyPatch):
+    monkeypatch.setattr(AsyncClient, "get", return_mocked_resp)
+    monkeypatch.setattr(MockResponse, "raise_for_status", raise_http_status_error_404)
+
+    with raises(HTTPStatusError) as error:
+        treasury_transfers = await covalent._get_transfer_data("", "", 1)
+
+        assert not treasury_transfers
```

Comment:
It could be that the evaluation of `assert not treasury_transfers` is itself returning a `HTTPStatusError`, in which case this version of the test would pass (whereas it should probably not).

Said another way, you should be testing for the fact that `treasury_transfers = await covalent._get_transfer_data("", "", 1)` raises an `HTTPStatusError`, not that the whole raise block (as it currently is) is raising.